qdev_wrappers/tests/dataset/test_doNd.py

```diff
@@ -0,0 +1,165 @@
+"""
+These are the basic black box tests for the doNd functions.
+"""
+
+from qdev_wrappers.dataset.doNd import do0d, do1d, do2d
+from typing import Tuple, List, Optional
+from qcodes.instrument.parameter import Parameter
+from qcodes import config, new_experiment
+from qcodes.utils import validators
+
+import pytest
+
+config.user.mainfolder = "output"  # set ouput folder for doNd's
+new_experiment("doNd-tests", sample_name="no sample")
+
+
+@pytest.fixture()
+def _parameters():
+
+    _param = Parameter('simple_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1)
+
+    _paramComplex = Parameter('simple_complex_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1 + 1j,
+                   vals=validators.ComplexNumbers())
+
+    _param_set = Parameter('simple_setter_parameter',
+                       set_cmd=None,
+                       get_cmd=None)
+
+    return _param, _paramComplex, _param_set
+
+
+def _param_func(_p):
+    """
+    A private utility function.
+    """
+    _new_param = Parameter('modified_parameter',
+                           set_cmd= None,
+                           get_cmd= lambda: _p.get()*2)
+    return _new_param
+
+
+@pytest.fixture()
+def _param_callable(_parameters):
+    _param, _, _ = _parameters
+    return _param_func(_param)
+
+
+def test_param_callable(_param_callable):
```

Comment:
This is because I inserted the "assert" within the function itself just for sanity check in the first version. This check does not require a separate test and even can be completely omitted. I prefer to loose the test and move assert into the fixture.

Reply:
Fixtures should ideally not do assertions.

Reply:
See my answer above.